solr/solrj/src/test/org/apache/solr/client/solrj/impl/LBHttp2SolrClientTest.java

```diff
@@ -50,8 +52,8 @@ public void testLBHttp2SolrClientWithTheseParamNamesInTheUrl() {
 
     try (Http2SolrClient http2SolrClient =
             new Http2SolrClient.Builder(url).withTheseParamNamesInTheUrl(urlParamNames).build();
```

Comment:
perhaps some tests methods shouldn't care in theory what the httpSolrClient impl is exactly but it'd be nice to get one randomly or via parameterization even better.  Not sure if that'd be too difficult.

Reply:
Actually, 4/5 test methods here are using a mock client.  The `withTheseParamNamesInTheUrl` case is being tested with the clients' own unit tests.  I did add a random choice to `LBHttp2SolrClientIntegrationTest` (renamed).  I think this is pretty good coverage already.